ui/app/reducers.js

```diff
@@ -41,17 +41,26 @@ function rootReducer (state, action) {
   return state
 }
 
+window.logStateString = function (cb) {
+  let state = window.METAMASK_CACHED_LOG_STATE
+  const version = global.platform.getVersion()
+  const browser = window.navigator.userAgent
+  return global.platform.getPlatformInfo((err, platform) => {
+    if (err) {
+      return cb(err)
+    }
+    state.version = version
+    state.platform = platform
+    state.browser = browser
+    let stateString = JSON.stringify(state, removeSeedWords, 2)
+    return cb(null, stateString)
+  })
+}
+
 window.logState = function () {
-  const state = window.METAMASK_CACHED_LOG_STATE
-  let version
-  try {
-    version = global.platform.getVersion()
-  } catch (e) {
-    version = 'unable to load version.'
-  }
-  state.version = version
-  const stateString = JSON.stringify(state, removeSeedWords, 2)
-  return stateString
+  return window.logStateString((result) => {
+    console.log(result)
+  })
```

Comment:
It's a little sad to me that this will break `copy(logState())`. I guess the new way will be something like `logStateString((err, res) => { copy(res) })`?

Maybe we could add a boolean argument for copying, so I could do `logState(true)` to copy to clipboard in one go?

Reply:
done